sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestTimestamped.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject;
+
+import org.jdbi.v3.core.H2DatabaseRule;
+import org.jdbi.v3.core.StatementContext;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.sqlobject.customizers.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.customizers.Timestamped;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for the {@link Timestamped} annotation
+ */
+public class TestTimestamped {
+    public PersonDAO personDAO;
+
+    @Rule
+    public H2DatabaseRule h2DatabaseRule = new H2DatabaseRule();
+
+    @Before
+    public void beforeEach() {
+        h2DatabaseRule.getJdbi().installPlugin(new SqlObjectPlugin());
+        personDAO = h2DatabaseRule.getJdbi().onDemand(PersonDAO.class);
+        personDAO.createTable();
+    }
+
+    @Test
+    public void shouldInsertCreatedAndModifiedFields() {
+        Person p = new Person("John", "Phiri");
+        p.setId(1);
+        personDAO.insert(p);
+
+        Person found = personDAO.get(1);
+
+        assertThat(found.getCreated()).isNotNull();
+        assertThat(found.getModified()).isNotNull();
+
+        // This is one way we can get the binding information of the executed query
+        h2DatabaseRule.getJdbi().setTimingCollector((l, statementContext) -> {
```

Comment:
This timing collector needs to be configured before the `personDAO.insert()` call for the assertion to actually be tested. (applies to all 3 test methods)